File: ui-java/src/main/java/com/silwek/tools/ui/java/UICompat.java
package com.silwek.tools.ui.java;

import android.app.Dialog;
import android.content.Context;
import android.view.View;
import android.widget.Toast;

import androidx.annotation.StringRes;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;

public class UICompat {

    public static void showToast(
            Context context,
            @StringRes int messageRes
    ) {
        showToast(context, messageRes, Toast.LENGTH_SHORT);
    }

    public static void showToast(
            Context context,
            @StringRes int messageRes,
            int duration
    ) {
        UIExtCompat.Companion.showToast(context, messageRes, duration);
    }

    public static void showToast(Context context, String message) {
        showToast(context, message, Toast.LENGTH_SHORT);
    }

    public static void showToast(Context context, String message, int duration) {
        UIExtCompat.Companion.showToast(context, message, duration);
    }

    public static boolean hasOpenedDialogs(FragmentActivity activity) {
        return UIExtCompat.Companion.hasOpenedDialogs(activity);
    }

    public static void showKeyboard(View view) {
        showKeyboard(view, null);
    }

    public static void showKeyboard(View view, Dialog parentDialog) {
        UIExtCompat.Companion.showKeyboard(view, parentDialog);
    }

    public static void hideKeyboard(Fragment fragment) {
        UIExtCompat.Companion.hideKeyboard(fragment);
    }

    public static void hideKeyboard(View view) {
        hideKeyboard(view, null);
    }

    public static void hideKeyboard(View view, Dialog parentDialog) {
        UIExtCompat.Companion.hideKeyboard(view, parentDialog);
    }
}

File: ui/src/main/java/com/silwek/tools/ui/UICompat.kt
package com.silwek.tools.ui

import android.app.Dialog
import android.content.Context
import android.view.View
import android.widget.Toast
import androidx.annotation.StringRes
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity

class UICompat {
    companion object {
        fun showToast(
            context: Context?,
            @StringRes messageRes: Int
        ) {
            showToast(context, messageRes, Toast.LENGTH_SHORT)
        }

        fun showToast(
            context: Context?,
            @StringRes messageRes: Int,
            duration: Int = Toast.LENGTH_SHORT
        ) {
            if (context == null) return
            context.showToast(messageRes, duration)
        }

        fun showToast(context: Context?, message: String) {
            showToast(context, message, Toast.LENGTH_SHORT)
        }

        fun showToast(context: Context?, message: String, duration: Int = Toast.LENGTH_SHORT) {
            if (context == null) return
            context.showToast(message, duration)
        }

        fun hasOpenedDialogs(activity: FragmentActivity?): Boolean {
            if (activity == null) return false
            return activity.hasOpenedDialogs()
        }

        fun showKeyboard(view: View?, parentDialog: Dialog? = null) {
            if (view == null) return
            view.showKeyboard(parentDialog)
        }

        fun hideKeyboard(fragment: Fragment?) {
            if (fragment == null) return
            fragment.hideKeyboard()
        }

        fun hideKeyboard(view: View?, parentDialog: Dialog? = null) {
            if (view == null) return
            view.hideKeyboard(parentDialog)
        }
    }
}